src/back/glsl/mod.rs

```diff
@@ -751,6 +755,11 @@ impl<'a, W: Write> Writer<'a, W> {
         }
 
         write!(self.out, "]")?;
+
+        if let TypeInner::Array { base, size, .. } = self.module.types[base].inner {
```

Comment:
Let's avoid shadowing the same name as the same type. It makes code confusing wrt what actual values are used.
I filed https://github.com/rust-lang/rust-clippy/issues/3433#issuecomment-1094555963 upstream to get clippy helping with this guideline.